crypto/pkcs12/p12_npas.c

```diff
@@ -211,8 +211,6 @@ static int alg_get(const X509_ALGOR *alg, int *pnid, int *piter,
         if (pbe2 == NULL)
             goto done;
 
-        X509_ALGOR_get0(&aoid, &aparamtype, &aparam, pbe2->keyfunc);
```

Comment:
This change does not look correct to me. This call fills in the value of `aparamtype` and `aparam` which are later referred to when finding the `kdf` value below.

Reply:
please try `X509_ALGOR_get0(NULL, &aparamtype, &aparam, pbe2->keyfunc)` instead.

Reply:
Thanks. Fixed.

Reply:
@mattcaswell